"Open Tickets" widget done properly second attempt :) + y-axis length settings added

MindGas opened the pull request:
After multiple attempts and re-configuring my developing space, this pull request looks better than last one.

MindGas committed:
"Open Tickets" widget done properly

Now "Open Tickets" widget displays values depending on queues selected.

MindGas committed:
Delete Opentickets.php

MindGas committed:
Delete CalculateTotalsOpenTicketsTask.php

MindGas committed:
Sync with coencoppens main

MindGas committed:
Cloned coencoppens 1.4.0 branch

MindGas committed:
"Open Tickets" widget done properly

Now "Open Tickets" widget displays values depending on queues selected.

coencoppens commented:
Well, I agree there's probably something going wrong here ;-) GitHub even says I can't automatically merge the pull request..

There are a lot of changes in your commit there shouldn't be "re committed". Did you update your branch before submitting?

MindGas commented:
I copied your 1.4.0 branch and done changes on that. It was done nasty way by downloading your branch committing it to my branch and making changes to it. It shouldn't be so complicated. I'm doing something wrong. Will try again later.

coencoppens commented:
I think this article might be useful: [Synching a fork](https://help.github.com/articles/syncing-a-fork)

MindGas committed:
Revert "2 new widgets and sub-stringed long names"

This reverts commit b5bce728df1cf9d96338f9973c742c762933c9c5.

MindGas committed:
Trying to merge 1.4.0

MindGas committed:
trying to merge 2

MindGas committed:
"Open Tickets" widget done properly 2 attempt

Now "Open Tickets" widget displays values depending on queues selected.

MindGas committed:
"Open Tickets" widget done properly 3 attempt

Unneeded files deleted

MindGas committed:
Update .gitignore

MindGas committed:
Delete .project

MindGas committed:
Widget y-axis length settings

Added global app settings and functionality to set y-axis length for
Account, Queues and Resources widgets.

MindGas committed:
Merge branch 'newwidgets' of https://github.com/MindaugasMeskauskas/autotask-dashboards into newwidgets

MindGas committed:
Custom Widgets

Allows to add any number of different custom widgets to any dashboard.
It is useful if you want to add external widgets, websites, webcams or
anything you want to new widget.

MindGas commented:
One more update to my branch. It allows to add any number of custom widgets for external content. It is very basic, but might extend on it later. No bugs or issues found except few CSS related layout problems which I will update and fix in few moments.

MindGas commented:
found bug. working on it now.

coencoppens commented on Controller/CustomwidgetsController.php:
```diff
@@ -0,0 +1,102 @@
+			) {
+				
+				// prevents editing default widgets
+				if ( !empty($iCustomwidgetId) && $iCustomwidgetId > 13 ) {
```
What if we make additional widgets? Then id#14 will be created and this will need updating too.

MindGas replied:
unfortunately yes, there is few more lines where it will need to be adjusted, unless you have idea how it can be done dynamically?

coencoppens commented on Controller/CustomwidgetsController.php:
```diff
@@ -0,0 +1,102 @@
+					$this->Customwidget->id = $iCustomwidgetId;
+					
+					
+					/*$this->Customwidget->updateAll( $this->request->data['Customwidget']);
```
Can you remove commented blocks like this from the code? It doesn't serve any purpose to keep them.

MindGas replied:
Thanks for pointing this out.

coencoppens commented:
@MindaugasMeskauskas , thanks for your hard work :) I have a few questions though:
- Why the changes in logging etc? Those are changes we don't need, and it also makes it harder to merge.
- You've renamed database fields which isn't needed. That will force users to not only update the code but also run (another) mysql update. Better leave the fields as they are.
- https://github.com/coencoppens/autotask-dashboards/pull/20/files#r6941612
- https://github.com/coencoppens/autotask-dashboards/pull/20/files#r6941629

I think it would be best if you do a pull request "per feature", so one for the custom widgets, one for the open tickets etc. I don't know if the custom widgets is something we want to include in the plugin natively, I'll have to discuss this internally.

Don't get me wrong, I really appreciate your efforts, I just want to get it right :-)

MindGas commented:
There shouldn't be any changes in logging or anything else except in what is needed for widgets to function. Here could be some testing mistakes (leftovers) or merging issues. Code was merged from four 1.4.0 branch about 20 days ago. I will go through code again line by line if needed get rid of unneeded code or changes. In my case three custom widgets are required. 2 for webcam as you have seen in picture and 1 for website monitoring. It makes scene for me and might for others, but if dashboard is dedicated just for autotask there is nothing I can do. Except to keep changes for myself. I will split pull request to individuals when I will have time. In mean time if you have spotted anything else let me know. I dont have much spare time so progress is slow. Im still fixing bug
